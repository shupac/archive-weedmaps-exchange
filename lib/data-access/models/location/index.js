import { types } from 'mobx-state-tree';

const Location = types.model('LocationModel', {
  source: types.string,
  phoneNumber: types.string,
  name: types.string,
  email: types.string,
  deliveryInstructions: types.string,
  contactName: types.string,
  active: types.boolean,
  id: types.identifier,
  address: types.model('AddressModel', {
    territory: types.string,
    streetAddress: types.string,
    postalCode: types.string,
    longitude: types.number,
    latitude: types.number,
    country: types.string,
    city: types.string,
    id: types.string,
  }),
  licenses: types.array(
    types.model('LicenseModel', {
      number: types.string,
      licenseType: types.string,
      id: types.string,
    }),
  ),
});

export type AddressType = {
  territory: string,
  streetAddress: string,
  postalCode: string,
  longitude: number,
  latitude: number,
  country: string,
  city: string,
  id: string,
};

export type LicensesType = {
  number: string,
  licenseType: string,
  id: string,
};

export type LocationType = {
  source: string,
  phoneNumber: string,
  name: string,
  email: string,
  deliveryInstructions?: string,
  contactName: string,
  active: boolean,
  id: string,
  address: AddressType,
  licenses: LicensesType[],
};

export default Location;
