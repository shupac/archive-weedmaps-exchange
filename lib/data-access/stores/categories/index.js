import { types, getEnv } from 'mobx-state-tree';
import Department, { type DepartmentType } from '../../models/departments';

const transform = departments =>
  departments.map(department => {
    const { id, name, position, iconImage, avatarImage } = department;

    return Department.create({
      id,
      name,
      position,
      iconImageUrl: iconImage ? iconImage.mediumUrl : '',
      avatarImageUrl: avatarImage ? avatarImage.mediumUrl : '',
    });
  });

const CategoryStore = types
  .model('CategoryStore', {
    departments: types.array(Department),
  })
  .actions(self => ({
    async getDepartments(regionId) {
      try {
        const departments = await getEnv(self).client.fetch(
          `/buyer/regions/${regionId}/departments?include=avatar_image,icon_image`,
        );

        const transformed = transform(departments);

        self.setDepartments(transformed);
      } catch (e) {
        console.log(e);
      }
    },
    setDepartments(departments) {
      self.departments = departments;
    },
  }));

export type CategoryStoreType = {
  getDepartments: (regionId: string) => void,
  departments: Array<DepartmentType>,
};

export default CategoryStore;
