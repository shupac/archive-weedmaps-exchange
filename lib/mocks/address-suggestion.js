export const mockSuggestedAddresses = [
  {address: "41 Discovery Park Boulevard, Seattle, WA, USA", id: "Ei00MSBEaXNjb3ZlcnkgUGFyayBCb3VsZXZhcmQsIFNlYXR0bGUsIFdBLCBVU0E"},
  {address: "41 Discovery Drive, The Dalles, OR, USA", id: "Eic0MSBEaXNjb3ZlcnkgRHJpdmUsIFRoZSBEYWxsZXMsIE9SLCBVU0E"},
  {address: "41 Discovery Circle Northeast, Issaquah, WA, USA", id: "EjA0MSBEaXNjb3ZlcnkgQ2lyY2xlIE5vcnRoZWFzdCwgSXNzYXF1YWgsIFdBLCBVU0E"},
  {address: "41 Discovery Trail, Vancouver, WA, USA", id: "EiY0MSBEaXNjb3ZlcnkgVHJhaWwsIFZhbmNvdXZlciwgV0EsIFVTQQ"},
  {address: "41 Discovery Street, Walla Walla, WA, USA", id: "Eik0MSBEaXNjb3ZlcnkgU3RyZWV0LCBXYWxsYSBXYWxsYSwgV0EsIFVTQQ"},
]
