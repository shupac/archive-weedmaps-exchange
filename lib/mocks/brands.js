export const mockBrands = [
  {
    slug: 'gfarmalabs',
    name: 'GfarmaLabs',
    description:
      '<p><strong><ins>Our Mission:</ins></strong></p> <p><em>Leader of the American Cannabis Revolution</em></p> <p> </p> <p>G FarmaLabs has become a well known brand throughout the cannabis community with both cannabis patients and connoisseurs alike. G FarmaLabs is committed to applying science and global resources to design and develop market leading brands in the cannabis flower, extracts, and edible product areas.</p> <p><strong><ins>History:</ins></strong></p> <p>Family owned and operated, G Farmalabs was conceived in 2007 by husband and wife team Ata and Nicole Gonzalez. Forced to relocate to California in 2009 after the financial crisis known as “The Great Recession,” Ata and Nicole believed the end of marijuana prohibition was near. Upon arriving in California, Ata attended Oaksterdam University and rented their first farm in Northern California. Together they carefully planted and harvested their first batch of marijuana flowers. Nicole is a member of a fourth generation tobacco family and has a strong background in the cigar industry. In 2013, G FarmaLabs launched America’s first complete brand of cannabis products.</p> <p>In 2015, Ata and Nicole’s idea had now become reality. G FarmaBrands was born. G FarmaLabs is the world’s first cannabis company that dedicates their effort to following the traditional lessons adapted from the Tobacco Industry. G FarmaLabs infused the two commercial enterprises to create a variety of products in an emerging market, setting the trend for the future of cannabis by merging the two industries.</p> <p><strong><ins>Process:</ins></strong></p> <p>G FarmaLabs works closely with a team of professionals who designed the California based processing center and production facility called the “Innovation Center” located in Desert Hot Springs, California. They have adopted the phrase “Napa Valley of Cannabis” and focus on blending the finest Domestic Cultivated Cannabis. Consistently cultivated, G FarmaLabs uses the same secret strains, each carefully chosen to guarantee a consistent experience.</p> <p>Perfecting the recipe and process, the G FarmaLabs farmers cultivate, harvest, dry, and trim their seven strains making it possible for customers to experience the aroma and feel the passion and precision the team at G FarmaLabs dedicates to their line of products.</p> <p><strong><ins>Products:</ins></strong></p> <p>Offering a variety of products from pre-rolls to edibles, G FarmaLabs caters to a wide market of cannabis users. G Stiks, their line of prestigious, 100% indoor grown pre-rolled joints are grown, processed, and rolled by G FarmaLabs and are available in 6 different options depending on user preference. For people who prefer Indica, G FarmaLabs G Stiks offers “ROJO”, packaged in red symbolizing a potent form of medicine. Patients in medical marijuana states enjoy the highly desired “VIOLETA” G Stik, packaged in purple, similar to violet, symbolizing various things from royalty to wealth, magic and mystery. The “AMARILLO” Sativa G Stiks symbolize a yellow butterfly purposely chosen to create a euphoric feeling for patients who suffer from depression. The “ROSADO” Sativa G Stik symbolizing one of the most celebrated flowers, the Pink Lotus Flower and is designed to uplift the heart and mind. The “VERDE” G Stik symbolizes energy and healing and the “NARAN-JA” G Stik symbolizes the Orange Rose. G FarmaLabs’ hybrid G Stiks offer a solution for connoisseurs on the go who need a product that is ready to go. The “VERDE” blend promotes energy healing and stress relief.</p> <p>In addition to G Stiks, G FarmaLabs also offers a line of vapor products, injectors, and edibles. Their Liquid Gold vape battery pens support 1.5 grams and are available in Super Silver Haze, Silver Haze, Blue Diamond, Green Crack , Orange Kush, Red Diesel, Pink Lemonade, Blackberry Kush, White Widow, and Purple Haze. They also proudly offer Indica, Sativa, and Hybrid Red Cannabis Oil Injectors in 0.5-1 gram dosages as well as NUG RUN Cannabis Oil injectors that are currently only offered in a 0.5 gram dosages designed to allow users to inhale Cannabinoids while avoiding harmful smoke toxins. G FarmaLabs offers infused decadent edible chocolate bars and truffles and 100% raw and unfiltered honey.</p> <p>From the detail of their packaging to their meticulous formula, G FarmaLabs proudly manufactures their own products delivering only the highest quality cannabis.</p> <p><strong><ins>Networks:</ins></strong></p> <p>Recognized as a Leader in the American Cannabis Revolution, G FarmaLabs has influenced celebrities such as The Game and has been featured on ABC News 7 and 790 KABC Talk Radio. G FarmaLabs can be found in dispensaries throughout Northern California, San Bernardino, Los Angeles, Orange County, Southern California, and Washington.</p>',
    id: 'd054b34f-428e-4583-a2a4-27f4e5aea6f3',
  },
  {
    slug: 'thclear-co',
    name: 'THClear Co',
    description:
      '<p>We are the industry leader in solventless distillate. We currently produce 6 different products with our Thclear Distillate, with our infamous uncut Private Reserve Cartridge being the most popular. Our other amazing products include: Disposable Pens, Cartridges, Syringes, Honeypot Vials, and our funky 50/50!</p>',
    id: 'd60ca06a-89db-4584-99a8-1f9b9f427e42',
  },
];
