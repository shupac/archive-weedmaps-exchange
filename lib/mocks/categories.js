
export const mockCategories = [
  {
    "slug":"concentrates",
    "position":1,
    "name":"Concentrates",
    "id":"81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
    "iconImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "id":"c77396fe-4293-4911-805c-2cd3afb18949"
    },
    "avatarImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "id":"82c8a744-10c6-48c9-b5a0-3a90de6d2c2f"
    }
  },
  {
    "slug":"edibles",
    "position":3,
    "name":"Edibles",
    "id":"e306a27d-5076-4074-ab7a-21f6485ad4bb",
    "iconImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "id":"b252a9f9-f06e-4981-80db-cf74d6c7573a"
    },
    "avatarImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "id":"97cc29eb-0ce4-4b55-81ae-e1899b6dc7ce"
    }
  }
];