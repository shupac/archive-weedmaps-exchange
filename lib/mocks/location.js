export const mockLocations = [
  {
    source: "weedmaps",
    phoneNumber: "(303) 555-5555",
    name: "Andrew's Cryb",
    email: "asciametta@aol.com",
    deliveryInstructions: "TEST.",
    contactName: "Andre Smithh",
    active: true,
    id: "6039ad85-7be7-45ce-a5f9-3e802eeba1e5",
    address: {
      territory: "CA",
      streetAddress: "55 Discovery",
      postalCode: "92618",
      longitude: -117.7597545,
      latitude: 33.6618199,
      country: "us",
      city: "Irvine",
      id: "b393b3e0-0814-4352-aa63-95a4d07fe1af"
    },
    licenses: [
      {
        number: "THX1138",
        licenseType: "medical",
        id: "e5ae232c-9ae7-4e81-b6f6-67bd5562706f"
      }
    ]
  },
  {
    source: "exchange",
    phoneNumber: "(303) 555-5555",
    name: "Test Location",
    email: "joesmith@aol.com",
    deliveryInstructions: "Knock four times.",
    contactName: "Joe Smith",
    active: false,
    id: "7f98075f-a924-4606-a817-b6f99a61f289",
    address: {
      territory: "CA",
      streetAddress: "44 Discovery",
      postalCode: "92618",
      longitude: -117.756439,
      latitude: 33.66517839999999,
      country: "us",
      city: "Irvine",
      id: "3cdf8df1-4a9d-43cc-9aac-2548eb08dc58"
    },
    licenses: [
      {
        number: "THX1138",
        licenseType: "medical",
        id: "6a102cdd-e21b-46ef-b4e7-55cfcb6e3bcb"
      }
    ]
  },
];

export const mockParsedLocation = {
  name: 'ShowGrow Irvine',
  address: '824 East 17th Street,Irvine, CA',
  instructions: 'Buzz 12345',
  contact: 'John Doe',
  phone: '2139735232',
  email: 'john@showgrowirvine.com',
  licenses: [
    {
      id: '123456',
      licenseType: 'Adult-Use Retail',
      number: '7645-2347-8743-3786',
    },
    {
      id: '54321',
      licenseType: 'Medical Retail',
      number: '3453-3453-3453-5435',
    },
  ],
};

export const mockFormLocation = {
  name: 'ShowGrow Irvine',
  address: '824 East 17th Street,Irvine, CA',
  instructions: 'Buzz 12345',
  contactName: 'John Doe',
  phoneNumber: '2139735232',
  email: 'john@showgrowirvine.com',
  licenses: [
    {
      id: '123456',
      licenseType: 'Adult-Use Retail',
      number: '7645-2347-8743-3786',
    },
    {
      id: '54321',
      licenseType: 'Medical Retail',
      number: '3453-3453-3453-5435',
    },
  ],
};
