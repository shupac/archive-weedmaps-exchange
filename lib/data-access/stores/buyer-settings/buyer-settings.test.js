import { getSnapshot } from 'mobx-state-tree';
import { mockCategories } from 'lib/mocks/categories';
import { mockBrands } from 'lib/mocks/brands';
import { mockLocations } from 'lib/mocks/location';
import BuyerSettings from './';

describe('BuyerSettings Store', () => {
  it('can fetch data from categories endpoint', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue({ data: mockCategories }),
    };
    const buyerSettings = BuyerSettings.create({}, { client: mockFetchClient });
    await buyerSettings.getDepartments();
    expect(mockFetchClient.fetch).toHaveBeenCalledWith(
      `/buyer/departments?include=avatar_image,icon_image,categories`,
    );
    expect(getSnapshot(buyerSettings.departmentsData)).toMatchSnapshot();
  });

  it('should transform departments data', () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue({ data: mockCategories }),
    };
    const buyerSettings = BuyerSettings.create({}, { client: mockFetchClient });
    buyerSettings.setDepartmentsData(mockCategories);
    expect(buyerSettings.departments).toMatchSnapshot();
    expect(buyerSettings.departmentsLoading).toEqual(false);
  });

  it('should return availabilities', () => {
    const buyerSettings = BuyerSettings.create({});
    expect(buyerSettings.availabilities).toMatchSnapshot();
  });

  it('can fetch data from brands endpoint', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue({ data: mockBrands }),
    };
    const buyerSettings = BuyerSettings.create({}, { client: mockFetchClient });
    await buyerSettings.getBrands();
    expect(mockFetchClient.fetch).toHaveBeenCalledWith(`/buyer/brands`);
    expect(getSnapshot(buyerSettings.brands)).toMatchSnapshot();
  });

  it('can fetch data from locations endpoint', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue({ data: mockLocations }),
    };
    const buyerSettings = BuyerSettings.create({}, { client: mockFetchClient });
    await buyerSettings.getLocations();
    expect(mockFetchClient.fetch).toHaveBeenCalledWith(`/locations`);
    expect(getSnapshot(buyerSettings.locations)).toMatchSnapshot();
    expect(getSnapshot(buyerSettings.activeLocation)).toMatchSnapshot();
  });

  it('should catch errors', () => {
    const mockErrorClient = {
      fetch: () => {
        throw new Error('test');
      },
    };

    const log = jest.spyOn(global.console, 'log').mockReturnValue();

    const buyerSettings = BuyerSettings.create({}, { client: mockErrorClient });
    buyerSettings.getDepartments();
    expect(log).toHaveBeenCalledWith(new Error('test'));
    buyerSettings.getBrands();
    expect(log).toHaveBeenCalledWith(new Error('test'));
    buyerSettings.getLocations();
    expect(log).toHaveBeenCalledWith(new Error('test'));

    log.mockRestore();
  });
  describe('when updating active location', () => {
    it('will push new location when not active already', async () => {
      const mockFetchClient = {
        fetch: jest.fn().mockReturnValue({ data: mockLocations }),
      };
      const buyerSettings = BuyerSettings.create(
        { locations: mockLocations },
        { client: mockFetchClient },
      );
      buyerSettings.updateActiveLocation(
        '7f98075f-a924-4606-a817-b6f99a61f289',
      );
      expect(buyerSettings.activeLocation).toEqual(mockLocations[1]);
    });
    it('will not push new location when active already', async () => {
      const mockFetchClient = {
        fetch: jest.fn().mockReturnValue({ data: mockLocations }),
      };
      const buyerSettings = BuyerSettings.create(
        { locations: mockLocations },
        { client: mockFetchClient },
      );
      const mocksetActiveLocation = jest.spyOn(
        buyerSettings,
        'setActiveLocation',
      );
      buyerSettings.updateActiveLocation(
        '6039ad85-7be7-45ce-a5f9-3e802eeba1e5',
      );
      expect(mocksetActiveLocation).not.toHaveBeenCalled();
    });
  });
});
