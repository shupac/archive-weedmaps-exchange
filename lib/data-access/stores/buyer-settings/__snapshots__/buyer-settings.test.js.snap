// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyerSettings Store can fetch data from brands endpoint 1`] = `
Array [
  Object {
    "description": "<p><strong><ins>Our Mission:</ins></strong></p> <p><em>Leader of the American Cannabis Revolution</em></p> <p> </p> <p>G FarmaLabs has become a well known brand throughout the cannabis community with both cannabis patients and connoisseurs alike. G FarmaLabs is committed to applying science and global resources to design and develop market leading brands in the cannabis flower, extracts, and edible product areas.</p> <p><strong><ins>History:</ins></strong></p> <p>Family owned and operated, G Farmalabs was conceived in 2007 by husband and wife team Ata and Nicole Gonzalez. Forced to relocate to California in 2009 after the financial crisis known as “The Great Recession,” Ata and Nicole believed the end of marijuana prohibition was near. Upon arriving in California, Ata attended Oaksterdam University and rented their first farm in Northern California. Together they carefully planted and harvested their first batch of marijuana flowers. Nicole is a member of a fourth generation tobacco family and has a strong background in the cigar industry. In 2013, G FarmaLabs launched America’s first complete brand of cannabis products.</p> <p>In 2015, Ata and Nicole’s idea had now become reality. G FarmaBrands was born. G FarmaLabs is the world’s first cannabis company that dedicates their effort to following the traditional lessons adapted from the Tobacco Industry. G FarmaLabs infused the two commercial enterprises to create a variety of products in an emerging market, setting the trend for the future of cannabis by merging the two industries.</p> <p><strong><ins>Process:</ins></strong></p> <p>G FarmaLabs works closely with a team of professionals who designed the California based processing center and production facility called the “Innovation Center” located in Desert Hot Springs, California. They have adopted the phrase “Napa Valley of Cannabis” and focus on blending the finest Domestic Cultivated Cannabis. Consistently cultivated, G FarmaLabs uses the same secret strains, each carefully chosen to guarantee a consistent experience.</p> <p>Perfecting the recipe and process, the G FarmaLabs farmers cultivate, harvest, dry, and trim their seven strains making it possible for customers to experience the aroma and feel the passion and precision the team at G FarmaLabs dedicates to their line of products.</p> <p><strong><ins>Products:</ins></strong></p> <p>Offering a variety of products from pre-rolls to edibles, G FarmaLabs caters to a wide market of cannabis users. G Stiks, their line of prestigious, 100% indoor grown pre-rolled joints are grown, processed, and rolled by G FarmaLabs and are available in 6 different options depending on user preference. For people who prefer Indica, G FarmaLabs G Stiks offers “ROJO”, packaged in red symbolizing a potent form of medicine. Patients in medical marijuana states enjoy the highly desired “VIOLETA” G Stik, packaged in purple, similar to violet, symbolizing various things from royalty to wealth, magic and mystery. The “AMARILLO” Sativa G Stiks symbolize a yellow butterfly purposely chosen to create a euphoric feeling for patients who suffer from depression. The “ROSADO” Sativa G Stik symbolizing one of the most celebrated flowers, the Pink Lotus Flower and is designed to uplift the heart and mind. The “VERDE” G Stik symbolizes energy and healing and the “NARAN-JA” G Stik symbolizes the Orange Rose. G FarmaLabs’ hybrid G Stiks offer a solution for connoisseurs on the go who need a product that is ready to go. The “VERDE” blend promotes energy healing and stress relief.</p> <p>In addition to G Stiks, G FarmaLabs also offers a line of vapor products, injectors, and edibles. Their Liquid Gold vape battery pens support 1.5 grams and are available in Super Silver Haze, Silver Haze, Blue Diamond, Green Crack , Orange Kush, Red Diesel, Pink Lemonade, Blackberry Kush, White Widow, and Purple Haze. They also proudly offer Indica, Sativa, and Hybrid Red Cannabis Oil Injectors in 0.5-1 gram dosages as well as NUG RUN Cannabis Oil injectors that are currently only offered in a 0.5 gram dosages designed to allow users to inhale Cannabinoids while avoiding harmful smoke toxins. G FarmaLabs offers infused decadent edible chocolate bars and truffles and 100% raw and unfiltered honey.</p> <p>From the detail of their packaging to their meticulous formula, G FarmaLabs proudly manufactures their own products delivering only the highest quality cannabis.</p> <p><strong><ins>Networks:</ins></strong></p> <p>Recognized as a Leader in the American Cannabis Revolution, G FarmaLabs has influenced celebrities such as The Game and has been featured on ABC News 7 and 790 KABC Talk Radio. G FarmaLabs can be found in dispensaries throughout Northern California, San Bernardino, Los Angeles, Orange County, Southern California, and Washington.</p>",
    "id": "d054b34f-428e-4583-a2a4-27f4e5aea6f3",
    "licenses": null,
    "name": "GfarmaLabs",
    "slug": "gfarmalabs",
  },
  Object {
    "description": "<p>We are the industry leader in solventless distillate. We currently produce 6 different products with our Thclear Distillate, with our infamous uncut Private Reserve Cartridge being the most popular. Our other amazing products include: Disposable Pens, Cartridges, Syringes, Honeypot Vials, and our funky 50/50!</p>",
    "id": "d60ca06a-89db-4584-99a8-1f9b9f427e42",
    "licenses": null,
    "name": "THClear Co",
    "slug": "thclear-co",
  },
]
`;

exports[`BuyerSettings Store can fetch data from categories endpoint 1`] = `
Array [
  Object {
    "avatarImage": Object {
      "id": "82c8a744-10c6-48c9-b5a0-3a90de6d2c2f",
      "largeUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "mediumUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "smallUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
    },
    "categories": Array [
      Object {
        "id": "c922a456-146e-4293-955b-bb517d3063ac",
        "name": "Sauce",
        "slug": "sauce",
      },
      Object {
        "id": "b7295924-975c-4e0c-bb18-250d4b233a35",
        "name": "Live Resin",
        "slug": "live-resin",
      },
      Object {
        "id": "2e6d3214-fac6-45a8-9786-6dbda76a6a33",
        "name": "Hash Oil",
        "slug": "hash-oil",
      },
      Object {
        "id": "89af476a-0f1e-4158-9781-ac3109f3c96c",
        "name": "Solventless",
        "slug": "solventless",
      },
      Object {
        "id": "0128b3c7-54d6-4b27-812d-718fc8ab4b4a",
        "name": "Ice Hash",
        "slug": "ice-hash",
      },
      Object {
        "id": "68ea1a1c-9a5c-44a8-b97e-daceb020c284",
        "name": "Dry Sift",
        "slug": "dry-sift",
      },
      Object {
        "id": "5e9573a9-a0b2-4b7e-b345-ce750ba62a41",
        "name": "Apparel",
        "slug": "apparel",
      },
      Object {
        "id": "bfae7a36-f12f-4faf-9a63-1d4e9b60d9be",
        "name": "Rosin",
        "slug": "rosin",
      },
      Object {
        "id": "20eef949-8169-4a80-a80e-8482a393e29d",
        "name": "Shatter",
        "slug": "shatter",
      },
      Object {
        "id": "934e604b-7ce6-4545-a61d-c44c32072a44",
        "name": "Distillate",
        "slug": "distillate",
      },
      Object {
        "id": "335bc5dd-15b1-4be8-8595-17cc1df3e853",
        "name": "Crumble",
        "slug": "crumble",
      },
      Object {
        "id": "6c56279f-be48-44be-8a53-8fb8303e154c",
        "name": "Budder",
        "slug": "budder",
      },
    ],
    "iconImage": Object {
      "id": "c77396fe-4293-4911-805c-2cd3afb18949",
      "largeUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "mediumUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "smallUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
    },
    "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
    "name": "Concentrates",
    "position": 1,
    "slug": "concentrates",
  },
  Object {
    "avatarImage": Object {
      "id": "97cc29eb-0ce4-4b55-81ae-e1899b6dc7ce",
      "largeUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "mediumUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "smallUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
    },
    "categories": Array [
      Object {
        "id": "963d5e18-0feb-4263-9bd4-2ebd1e35693b",
        "name": "Beverages",
        "slug": "beverages",
      },
      Object {
        "id": "f245a838-fd25-4c7d-af45-e989c6702304",
        "name": "Munchies",
        "slug": "munchies",
      },
      Object {
        "id": "bf045477-c46d-4f12-b1bf-94f0f04e7f33",
        "name": "Baked Goods",
        "slug": "baked-goods",
      },
      Object {
        "id": "1e9b1182-59e6-4b9a-92ae-2c731a517409",
        "name": "Candies",
        "slug": "candies",
      },
      Object {
        "id": "95f27620-bc0f-421f-8d6f-d0e6abb1ec4c",
        "name": "Cooking Ingredients",
        "slug": "cooking-ingredients",
      },
      Object {
        "id": "af3eeab3-26f9-425b-99b5-fa70021e0d45",
        "name": "Gummies",
        "slug": "gummies",
      },
      Object {
        "id": "330979e6-bacf-42e0-adbd-bf623c590a7a",
        "name": "Chocolate",
        "slug": "chocolate",
      },
    ],
    "iconImage": Object {
      "id": "b252a9f9-f06e-4981-80db-cf74d6c7573a",
      "largeUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "mediumUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "smallUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
    },
    "id": "e306a27d-5076-4074-ab7a-21f6485ad4bb",
    "name": "Edibles",
    "position": 3,
    "slug": "edibles",
  },
]
`;

exports[`BuyerSettings Store can fetch data from locations endpoint 1`] = `
Array [
  Object {
    "active": true,
    "address": Object {
      "city": "Irvine",
      "country": "US",
      "id": "b393b3e0-0814-4352-aa63-95a4d07fe1af",
      "latitude": 33.6618199,
      "longitude": -117.7597545,
      "postalCode": "92618",
      "streetAddress": "55 Discovery",
      "territory": "CA",
    },
    "contactName": "Andre Smithh",
    "deliveryInstructions": "TEST.",
    "email": "asciametta@aol.com",
    "id": "6039ad85-7be7-45ce-a5f9-3e802eeba1e5",
    "licenses": Array [
      Object {
        "id": "e5ae232c-9ae7-4e81-b6f6-67bd5562706f",
        "licenseType": "medical",
        "number": "THX1138",
      },
    ],
    "name": "Andrew's Cryb",
    "phoneNumber": "(303) 555-5555",
    "source": "weedmaps",
  },
  Object {
    "active": false,
    "address": Object {
      "city": "Irvine",
      "country": "US",
      "id": "3cdf8df1-4a9d-43cc-9aac-2548eb08dc58",
      "latitude": 33.66517839999999,
      "longitude": -117.756439,
      "postalCode": "92618",
      "streetAddress": "44 Discovery",
      "territory": "CA",
    },
    "contactName": "Joe Smith",
    "deliveryInstructions": "Knock four times.",
    "email": "joesmith@aol.com",
    "id": "7f98075f-a924-4606-a817-b6f99a61f289",
    "licenses": Array [
      Object {
        "id": "6a102cdd-e21b-46ef-b4e7-55cfcb6e3bcb",
        "licenseType": "medical",
        "number": "THX1138",
      },
    ],
    "name": "Test Location",
    "phoneNumber": "(303) 555-5555",
    "source": "exchange",
  },
]
`;

exports[`BuyerSettings Store should return availabilities 1`] = `
Array [
  Object {
    "id": "inStock",
    "name": "In Stock",
  },
  Object {
    "id": "outOfStock",
    "name": "Out of Stock",
  },
]
`;

exports[`BuyerSettings Store should transform departments data 1`] = `
Array [
  Object {
    "avatarImageUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
    "categories": Array [
      Object {
        "id": "c922a456-146e-4293-955b-bb517d3063ac",
        "name": "Sauce",
        "slug": "sauce",
      },
      Object {
        "id": "b7295924-975c-4e0c-bb18-250d4b233a35",
        "name": "Live Resin",
        "slug": "live-resin",
      },
      Object {
        "id": "2e6d3214-fac6-45a8-9786-6dbda76a6a33",
        "name": "Hash Oil",
        "slug": "hash-oil",
      },
      Object {
        "id": "89af476a-0f1e-4158-9781-ac3109f3c96c",
        "name": "Solventless",
        "slug": "solventless",
      },
      Object {
        "id": "0128b3c7-54d6-4b27-812d-718fc8ab4b4a",
        "name": "Ice Hash",
        "slug": "ice-hash",
      },
      Object {
        "id": "68ea1a1c-9a5c-44a8-b97e-daceb020c284",
        "name": "Dry Sift",
        "slug": "dry-sift",
      },
      Object {
        "id": "5e9573a9-a0b2-4b7e-b345-ce750ba62a41",
        "name": "Apparel",
        "slug": "apparel",
      },
      Object {
        "id": "bfae7a36-f12f-4faf-9a63-1d4e9b60d9be",
        "name": "Rosin",
        "slug": "rosin",
      },
      Object {
        "id": "20eef949-8169-4a80-a80e-8482a393e29d",
        "name": "Shatter",
        "slug": "shatter",
      },
      Object {
        "id": "934e604b-7ce6-4545-a61d-c44c32072a44",
        "name": "Distillate",
        "slug": "distillate",
      },
      Object {
        "id": "335bc5dd-15b1-4be8-8595-17cc1df3e853",
        "name": "Crumble",
        "slug": "crumble",
      },
      Object {
        "id": "6c56279f-be48-44be-8a53-8fb8303e154c",
        "name": "Budder",
        "slug": "budder",
      },
    ],
    "iconImageUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
    "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
    "name": "Concentrates",
  },
  Object {
    "avatarImageUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
    "categories": Array [
      Object {
        "id": "963d5e18-0feb-4263-9bd4-2ebd1e35693b",
        "name": "Beverages",
        "slug": "beverages",
      },
      Object {
        "id": "f245a838-fd25-4c7d-af45-e989c6702304",
        "name": "Munchies",
        "slug": "munchies",
      },
      Object {
        "id": "bf045477-c46d-4f12-b1bf-94f0f04e7f33",
        "name": "Baked Goods",
        "slug": "baked-goods",
      },
      Object {
        "id": "1e9b1182-59e6-4b9a-92ae-2c731a517409",
        "name": "Candies",
        "slug": "candies",
      },
      Object {
        "id": "95f27620-bc0f-421f-8d6f-d0e6abb1ec4c",
        "name": "Cooking Ingredients",
        "slug": "cooking-ingredients",
      },
      Object {
        "id": "af3eeab3-26f9-425b-99b5-fa70021e0d45",
        "name": "Gummies",
        "slug": "gummies",
      },
      Object {
        "id": "330979e6-bacf-42e0-adbd-bf623c590a7a",
        "name": "Chocolate",
        "slug": "chocolate",
      },
    ],
    "iconImageUrl": "https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
    "id": "e306a27d-5076-4074-ab7a-21f6485ad4bb",
    "name": "Edibles",
  },
]
`;
